tests/UnitTests/Foundation/DispatchTests.mm

```diff
@@ -0,0 +1,200 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <TestFramework.h>
+#import <Foundation/Foundation.h>
+#import <dispatch/dispatch.h>
+
+static const char* key = "Specific Key";
+static const char* key2 = "Specific Key2";
+static const char* key3 = "Specific Key3";
+static char* context = "Specific Context 1";
+static char* context2 = "Specific Context 2";
+static char* context3 = "Specific Context 3";
+static int destructorCount = 0;
+
+// Create a 1 second end time for the test to timeout.
+static dispatch_time_t _createEndTestTime() {
+	return dispatch_time(DISPATCH_TIME_NOW, 1000000000000);
+}
+
+static void destructor(void* x){
+    destructorCount++;
+}
+
+TEST(DispatchTests, SimpleSpecific) {
+    dispatch_queue_t queue = dispatch_queue_create("queue", nullptr);
+	dispatch_group_t group = dispatch_group_create();
+
+	dispatch_queue_set_specific(queue, key, context, nullptr);
+
+	dispatch_group_async(group, queue, ^{
+		EXPECT_EQ(dispatch_get_specific(key), context);
+        dispatch_queue_set_specific(queue, key, context2, nullptr);
```

Comment:
The formatting  in this test file seems all sorts of strange. Is that just GitHub acting up?

Reply:
Yeah, it seems to just be github acting strangely. The format in the file itself seems fine to me.